src/main/scala/org/ergoplatform/nodeView/mempool/ErgoMemPool.scala

```diff
@@ -7,14 +7,61 @@ import org.ergoplatform.nodeView.state.{ErgoState, UtxoState}
 import org.ergoplatform.settings.ErgoSettings
 import scorex.core.transaction.MemoryPool
 import scorex.core.transaction.state.TransactionValidation
-import scorex.util.ModifierId
+import scorex.util.{ModifierId, bytesToId}
 
 import scala.util.Try
 
+object MemPoolStatisticsParams {
+  val nHistogramBins = 60  /* one hour */
+  val measurementIntervalMsec = 60 * 1000 /* one hour */
+}
+
+case class FeeHistogramBin(var nTxns: Int, var totalFee: Long)
+
+case class MemPoolStatistics() {
+  var startMeasurement: Long = System.currentTimeMillis() // start of measurement interval
+  var takenTxns : Long = 0              // amount of taken transaction since start of measurement
+  var snapTime: Long = startMeasurement // last snapshot time
+  var snapTakenTxns : Long = 0          // amount of transaction at the moment of last snapshot
+  val histogram = Array.fill(MemPoolStatisticsParams.nHistogramBins)(FeeHistogramBin(0,0))
+
+  //                  <.............takenTxns...................
+  //                                        <....snapTakenTxns..
+  // ------------------------------------------------------------> time
+  //                  ^                     ^                  ^
+  //                  |<measurementInterval>|                  |
+  // startMeasurement +            snapTime +      current time+
+
+  /**
+    * Add new entry to mempool statistics. This method is called when transaction is taken from mempool
+    * and placed in blockchain. It is not called when transaction in thrown away from the pool by replacement policy.
+    * To make statistic better represent most recent system behaviour, we periodically (each measurementIntervalMsec)
+    * prune statistic. To avoid situtation when we do not have statistic at all, we actually keep data up to
+    * 2*measurementIntervalMsec and periodically cut half of range.
+    */
+  def add(wtx : WeightedTxId) : Unit = {
+    val now = System.currentTimeMillis()
```

Comment:
The time provider can be introduced to abstract from time source and make this class unit-testable.